libs/libvtrutil/src/vtr_expr_eval.cpp

```diff
@@ -2,8 +2,14 @@
 #include "vtr_error.h"
 #include "vtr_util.h"
 #include "vtr_math.h"
+
 #include <string>
 #include <sstream>
+#include <iostream>
+
+current_information current_info_e;
```

Comment:
After looking through the code, I think you can remove these global variables and just pass in a struct or class to the expression evaluator with this additional info.

Reply:
Add a high-level comment of what this file is for: expression evaluation, with following operators (list them) supported; this can reduce expressions to numbers and is useful in architecture files. It can also check the state (passed in) of some variables in the placer and router for breakpoint evaluation.

Reply:
resolved